bike_kollective/lib/ui/screens/ride_feedback/ride_feedback_screen.dart

```diff
@@ -50,32 +49,47 @@ class _RideFeedbackScreenState extends State<RideFeedbackScreen> {
     });
   }
 
-  void submitFeedback() {
-    if (selectedStars != null) {
+  Future<void> submitFeedback() async {
+    if (_formKey.currentState!.validate() && selectedStars != null) {
       final review = RideReview(
         stars: selectedStars!,
         tags: likedTags + improvementTags, // Combine the tags
         comment: commentController.text,
         submitted: DateTime.now(),
       );
 
-
-      // TODO: Implement logic to save the review to database
-      // await Firestore.instance.collection('rides').document(widget.ride.docRef.id).update({
-      //   'review': review.toMap(),
-      // });
-      // print('Feedback submitted: ${review.toMap()}');
-      print('Feedback submitted');
-      
-   // Navigate back after submission
-      Navigator.pop(context);
+      // Save the review to the database
+      try {
+        await FirebaseFirestore.instance
```

Comment:
Use the Riverpod provider for the active_ride to provide the review/finish the ride

Reply:
ok.I think I implemented it correctly.